proposals/interceptors-mvp.md

```diff
@@ -0,0 +1,32 @@
+Interceptors MVP:
```

Comment:
Why is this a separate doc vs. being part of the other doc?

Reply:
This basically was a way for me to quickly get some thoughts out, I didn't actually think about the fact that I was pushing to a branch which was already being PR'ed, I'm intending to "collapse" this doc back into the single doc and then open back up for review.